utils.js

```diff
@@ -585,21 +585,38 @@ function makeDefaults(html, userID) {
   };
 }
 
-function parseAndCheckLogin(jar, defaultFuncs) {
+function parseAndCheckLogin(jar, defaultFuncs, retryCount) {
+  if (retryCount == undefined) {
+    retryCount = 0;
+  }
   return function(data) {
     return bluebird.try(function() {
       log.verbose("parseAndCheckLogin", data.body);
       if (data.statusCode >= 500 && data.statusCode < 600) {
-        log.warn("parseAndCheckLogin", "Got status code " + data.statusCode + " retrying...");
+        if (retryCount == 5) {
```

Comment:
Can you make this if `>= 5` in case something funny ever happens so we don't end up in an infinite loop?